### ISSUE TITLE
The training time?

### ISSUE THREAD

**Issue by wenyuqing** [HHYHRHY/MM-ACT#10]
Hello, thanks for your good job! How long do you take to train on Libero or Robotwin with 8 A100 GPUs? Because I wonder if it's possible to reproduce your work quickly with limited compute resource.

**Comment by HHYHRHY** [comment]
To reproduce our best results on Libero using 8x A100 GPUs, it typically takes about 48 hours per subset (training for approximately 10k gradient steps).
If you are looking to reduce the training time, you can try adjusting the batch_size to 64.

**Comment by wenyuqing** [comment]
Thanks for your reply! Is this time including all training stages or just a single stage?

**Comment by HHYHRHY** [comment]
This time refers to the training setting for pure action generation. For the multimodal learning pipeline, Stage 1 typically completes within a few hundred steps, usually taking around 3 hours to train. For Stage 2, the number of total gradient steps is the same as in the pure-action training, but since the gradients at each step come from multiple modalities (not just action), the total batch size increases proportionally. Therefore, more parallel GPUs or higher gradient accumulation steps are required to achieve this.

**Comment by wenyuqing** [comment]
I try to run configs/mmact_libero_action.yaml on 8 h20 GPUs on libero_goal dataset and set the epoch to 20, the bs is 4x4x8=128.  
The estimated time is just about 2hours, but the actual time is much longer than this(more than 1-2 days). Is this caused by this warning message? Have you encountered this issue?

<img width="1250" height="269" alt="Image" src="https://github.com/user-attachments/assets/6524c03f-7a74-47f8-bdc5-b8a96dbd0b72" />

**Comment by HHYHRHY** [comment]
2 hours  is the time for training one epoch. If the "actual time" you mentioned refers to the total training time, then it would be 2 hours multiplied by 20 epochs. This warning does not affect the training process. You can check the train.log in the output directory to see the specific timestamps.

**Comment by wenyuqing** [comment]
> 2 hours is the time for training one epoch. If the "actual time" you mentioned refers to the total training time, then it would be 2 hours multiplied by 20 epochs. This warning does not affect the training process. You can check the train.log in the output directory to see the specific timestamps.

Thanks for your quick reply! And I have another question. Does the shared-modality training mean the model output all three modalities together in the forward process or it just output single modality?

**Comment by HHYHRHY** [comment]
You can refer to https://github.com/HHYHRHY/MM-ACT/issues/9 for this question.

**Comment by HHYHRHY** [comment]
For training time, you can refer to the latest commit https://github.com/HHYHRHY/MM-ACT/commit/8f77ce73d5cb1d62178a4ba73e3270b32e71ae32, updating the accelerate config for faster training.
